USWDS-Compile - Bug: pnpm support

### Describe the bug

Using `gulp-compile` within a `pnpm` workspace fails in the `buildSass` function. `pnpm` is a package manager that has a local package cache with a unique directory structure.


### Steps to reproduce the bug

1. Use gulpfile.js example from documentation
2. `pnpm gulp build`


### Expected Behavior

Same as via `npx gulp build`

### Related code

The path resolved here does not exist: https://github.com/uswds/uswds-compile/blob/cd6f8dddddf5155b065fcbe3999ed8f4f89e2e8f/gulpfile.js#L159


### Screenshots

The corresponding exception is:

```
[14:05:47] 'buildSass' errored after 536 μs
[14:05:47] Error: Cannot find module '../../uswds/package.json'
Require stack:
- /Users/danielnaab/src/10x/atj-platform/node_modules/.pnpm/@uswds+compile@1.0.0_ts-node@10.9.1/node_modules/@uswds/compile/gulpfile.js
- /Users/danielnaab/src/10x/atj-platform/packages/design/gulpfile.js
- /Users/danielnaab/src/10x/atj-platform/node_modules/.pnpm/gulp-cli@2.3.0/node_modules/gulp-cli/lib/shared/require-or-import.js
- /Users/danielnaab/src/10x/atj-platform/node_modules/.pnpm/gulp-cli@2.3.0/node_modules/gulp-cli/lib/versioned/^4.0.0/index.js
- /Users/danielnaab/src/10x/atj-platform/node_modules/.pnpm/gulp-cli@2.3.0/node_modules/gulp-cli/index.js
- /Users/danielnaab/src/10x/atj-platform/node_modules/.pnpm/gulp@4.0.2/node_modules/gulp/bin/gulp.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
    at Module._load (node:internal/modules/cjs/loader:920:27)
    at Module.require (node:internal/modules/cjs/loader:1141:19)
    at require (node:internal/modules/cjs/helpers:110:18)
    at buildSass (/Users/danielnaab/src/10x/atj-platform/node_modules/.pnpm/@uswds+compile@1.0.0_ts-node@10.9.1/node_modules/@uswds/compile/gulpfile.js:159:15)
    at bound (node:domain:433:15)
    at runBound (node:domain:444:12)
    at asyncRunner (/Users/danielnaab/src/10x/atj-platform/node_modules/.pnpm/async-done@1.3.2/node_modules/async-done/index.js:55:18)
    at process.processTicksAndRejections (node:internal/process/task_queues:77:11)
[14:05:47] 'compile' errored after 2.47 ms
 ELIFECYCLE  Command failed with exit code 1.
```

### System setup

- USWDS Compile version: 1.0.0
- USWDS version: 3.6.0
- Device: m1 Macbook Pro
- Operating system: macOS 13.5.2
- Browser and version: n/a


### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-compile/issues) for duplicate bug reports.

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -162,11 +162,6 @@ function getUswdsVersion() {
 }
 
 function buildSass() {
-  let uswdsPath = "uswds";
-  if (settings.version === 3) {
-    uswdsPath = "@uswds/uswds";
-  }
-
   const pkg = getUswdsVersion();
 
   log(colors.blue, `Compiling with USWDS ${pkg}`);

--- a/gulpfile.js
+++ b/gulpfile.js
@@ -1,6 +1,7 @@
 const autoprefixer = require("autoprefixer");
 const csso = require("postcss-csso");
 const { src, dest, series, parallel, watch } = require("gulp");
+const path = require("path");
 const postcss = require("gulp-postcss");
 const replace = require("gulp-replace");
 const sass = require("gulp-sass")(require("sass-embedded"));
@@ -150,13 +151,23 @@ function logVersion() {
   return Promise.resolve("logged version");
 }
 
+function getUswdsVersion() {
+  let uswdsPackage = "uswds";
+  if (settings.version === 3) {
+    uswdsPackage = "@uswds/uswds";
+  }
+  const packagePath = path.join(path.dirname(require.resolve(uswdsPackage)), '../../');
+  const version = require(`${packagePath}/package.json`).version;
+  return version;
+}
+
 function buildSass() {
   let uswdsPath = "uswds";
   if (settings.version === 3) {
     uswdsPath = "@uswds/uswds";
   }
 
-  const pkg = require(`../../${uswdsPath}/package.json`).version;
+  const pkg = getUswdsVersion();
 
   log(colors.blue, `Compiling with USWDS ${pkg}`);
   const buildSettings = {

